avocado/utils/path.py

```diff
@@ -93,14 +96,23 @@ def find_command(cmd, default=None, check_exec=True):
     common_bin_paths = ["/usr/libexec", "/usr/local/sbin", "/usr/local/bin",
                         "/usr/sbin", "/usr/bin", "/sbin", "/bin"]
     try:
-        path_paths = os.environ['PATH'].split(":")
+        path_paths = None
+        if session:
+            path_paths = session.cmd("echo $PATH").stdout_text.strip('/n').split(":")
+        else:
+            path_paths = os.environ['PATH'].split(":")
     except IndexError:
         path_paths = []
     path_paths = list(set(common_bin_paths + path_paths))
 
     for dir_path in path_paths:
         cmd_path = os.path.join(dir_path, cmd)
-        if os.path.isfile(cmd_path):
+        if session and session.cmd("test -f %s" % cmd_path).exit_status == 0:
+            if check_exec:
+                if session.cmd("test -x %s" % cmd_path).exit_status != 0 or session.cmd("test -r %s" % cmd_path).exit_status != 0:
+                    continue
+            return session.cmd("realpath %s" % cmd_path).stdout_text.rstrip()
```

Comment:
Not sure if we want `realpath` since it is not built into the kernel. I'm open to other options.